File: python/PE030.py
#!/usr/bin/env python

"""
First, do a little math to estimate the range of the number: 
    I get to know that the numbers will not exceed 1000000
"""

pow5 = [i ** 5 for i in range(10)]

def fifthPower(n):
    global pow5
    return sum(map(lambda x: pow5[int(x)], str(n)))

def main():
    print sum(i for i in range(10, 1000000) if i == fifthPower(i))

if __name__ == '__main__':
    main()

File: java/PE030.java
public class PE030 {
    public static void main(String[] args) {
        int sum = 0;
        for(int i = 2; i <= 999999; i++) {
            if(isFifth(i)) {
                sum += i;
            }
        }
        System.out.println(sum);
    }
    private static boolean isFifth(int i) {
        int j = i;
        int sum = 0;
        while(j != 0) {
            sum += (int)Math.pow(j%10, 5);
            j /= 10;
        }
        if(sum == i) {
            return true;
        }
        else {
            return false;
        }
    }
}
